### PR TITLE
Proofreading fixes part 2 - Android basic security testing

### PR DESCRIPTION
**Pull request by Laancelot**
Links, typos

Thank you for submitting a Pull Request to the OWASP MASTG. Please make sure that:

- [ ] Your contribution is written in the 2nd person (e.g. you)
- [ ] Your contribution is written in an active present form for as much as possible.
- [ ] You have made sure that the reference section is up to date (e.g. please add sources you have used, make sure that the references to MITRE/MASVS/etc. are up to date)
- [ ] Your contribution has proper formatted markdown and/or code
- [ ] Any references to website have been formatted as [TEXT](URL “NAME”)
- [ ] You verified/tested the effectiveness of your contribution (e.g.: is the code really an effective remediation? Please verify it works!)

If your PR is related to an issue. Please end your PR test with the following line:
This PR closes #< insert number here >.

### CODE REVIEW

**Comment by Laancelot** [comment]
the URL checker failed on the URL that was changed in a previous pull request (Android basic security testing part 1) that hasn't been merged yet

**Review by Faby666 — COMMENTED** [review]
#2297

**Review by cpholguera — CHANGES_REQUESTED** [review]
Hi @Laancelot, thanks for adding missing links.

I got just one comment to that. 

Instead of:

```suggestion
 "[Recommended Tools - Objection](https://github.com/OWASP/owasp-mastg/blob/master/Document/0x08a-Testing-Tools.md#objection "Testing Tools - Objection")"
```

It should be:

```suggestion
 "[Recommended Tools - Objection](0x08a-Testing-Tools.md#objection "Testing Tools - Objection")"
```

Could you please correct them? Thank you very much!

**Comment by Laancelot** [comment]
done @cpholguera

**Comment by cpholguera** [comment]
I'm sorry @Laancelot but the GitHub rendering for the markdown block above was broken and it was showing a link instead of a block of code. Please take another look because the links are different. Thank you very much again!

File: Document/0x05b-Basic-Security_Testing.md [hunk 1]
```diff
@@ -200,7 +200,7 @@ If you're using a rooted device you can now start exploring the whole file syste
 
 This option is useful when you are working on a specific app and want to copy files you might encounter inside its sandbox (notice that you'll only have access to the files that the target app has access to). This approach works without having to set the app as debuggable, which is otherwise required when using Android Studio's Device File Explorer.
 
-First, connect to the app with Objection as explained in "Recommended Tools - Objection". Then, use `ls` and `cd` as you normally would on your terminal to explore the available files:
+First, connect to the app with Objection as explained in "[Recommended Tools - Objection](https://github.com/OWASP/owasp-mastg/pull/0x08a-Testing-Tools.md#objection "Testing Tools - Objection")". Then, use `ls` and `cd` as you normally would on your terminal to explore the available files:
```
Comment (cpholguera) [inline Document/0x05b-Basic-Security_Testing.md:203]: Thanks for adding missing links

```suggestion
First, connect to the app with Objection as explained in "[Recommended Tools - Objection](0x08a-Testing-Tools.md#objection "Testing Tools - Objection")". Then, use `ls` and `cd` as you normally would on your terminal to explore the available files:
```

File: Document/0x05b-Basic-Security_Testing.md [hunk 2]
```diff
@@ -640,7 +640,7 @@ Each folder has its own purpose:
 
 However, the app might store more data not only inside these folders but also in the parent folder (`/data/data/[package-name]`).
 
-Refer to the "Testing Data Storage" chapter for more information and best practices on securely storing sensitive data.
+Refer to the "[Testing Data Storage](https://github.com/OWASP/owasp-mastg/pull/0x05d-Testing-Data-Storage.md "Testing Data Storage")" chapter for more information and best practices on securely storing sensitive data.
```
Comment (cpholguera) [inline Document/0x05b-Basic-Security_Testing.md:643]:
```suggestion
Refer to the "[Testing Data Storage](0x05d-Testing-Data-Storage.md "Testing Data Storage")" chapter for more information and best practices on securely storing sensitive data.
```

File: Document/0x05b-Basic-Security_Testing.md [hunk 3]
```diff
@@ -517,10 +517,10 @@ The Android Manifest is the main source of information, it includes a lot of int
...
-- Backup allowance: `android:allowBackup` (see "Data Storage on Android")
-- App components: `activity`, `service`, `provider`, `receiver` (see "Android Platform APIs" and "Data Storage on Android")
-- Debuggable flag: `debuggable` (see "Code Quality and Build Settings of Android Apps")
+- App permissions: `permission` (see "[Android Platform APIs](https://github.com/OWASP/owasp-mastg/pull/0x05h-Testing-Platform-Interaction.md "Testing Platform Interaction")")
+- Backup allowance: `android:allowBackup` (see "[Data Storage on Android](https://github.com/OWASP/owasp-mastg/pull/0x05d-Testing-Data-Storage.md "Testing Data Storage)")
+- App components: `activity`, `service`, `provider`, `receiver` (see "[Android Platform APIs](https://github.com/OWASP/owasp-mastg/pull/0x05h-Testing-Platform-Interaction.md "Testing Platform Interaction")" and "[Data Storage on Android](https://github.com/OWASP/owasp-mastg/pull/0x05d-Testing-Data-Storage.md "Testing Data Storage)")
+- Debuggable flag: `debuggable` (see "[Code Quality and Build Settings of Android Apps](https://github.com/OWASP/owasp-mastg/pull/0x05i-Testing-Code-Quality-and-Build-Settings.md "Testing Code Quality and Build Settings")")
```
Comment (cpholguera) [inline Document/0x05b-Basic-Security_Testing.md:523]:
```suggestion
- App permissions: `permission` (see "[Android Platform APIs](https://github.com/OWASP/owasp-mastg/pull/0x05h-Testing-Platform-Interaction.md "Testing Platform Interaction")")
- Backup allowance: `android:allowBackup` (see "[Data Storage on Android](0x05d-Testing-Data-Storage.md "Testing Data Storage)")
- App components: `activity`, `service`, `provider`, `receiver` (see "[Android Platform APIs](0x05h-Testing-Platform-Interaction.md "Testing Platform Interaction")" and "[Data Storage on Android](0x05d-Testing-Data-Storage.md "Testing Data Storage)")
- Debuggable flag: `debuggable` (see "[Code Quality and Build Settings of Android Apps](0x05i-Testing-Code-Quality-and-Build-Settings.md "Testing Code Quality and Build Settings")")
```

**Review by cpholguera — CHANGES_REQUESTED** [review]
Sending the right links as suggestions, thanks @Laancelot!

File: Document/0x05b-Basic-Security_Testing.md [hunk 4]
```diff
@@ -517,10 +517,10 @@ The Android Manifest is the main source of information, it includes a lot of int
 
 Here's a non-exhaustive list of some info and the corresponding keywords that you can easily search for in the Android Manifest by just inspecting the file or by using `grep -i <keyword> AndroidManifest.xml`:
 
-- App permissions: `permission` (see "Android Platform APIs")
-- Backup allowance: `android:allowBackup` (see "Data Storage on Android")
-- App components: `activity`, `service`, `provider`, `receiver` (see "Android Platform APIs" and "Data Storage on Android")
-- Debuggable flag: `debuggable` (see "Code Quality and Build Settings of Android Apps")
+- App permissions: `permission` (see "[Android Platform APIs](https://github.com/OWASP/owasp-mastg/pull/0x05h-Testing-Platform-Interaction.md "Testing Platform Interaction")")
```
Comment (cpholguera) [inline Document/0x05b-Basic-Security_Testing.md:520]:
```suggestion
- App permissions: `permission` (see "[Android Platform APIs](0x05h-Testing-Platform-Interaction.md "Testing Platform Interaction")")
```

**Review by cpholguera — APPROVED** [review]
Thanks @Laancelot!

**Pull request merged**